libs/query-engine-common/src/engine.rs

```diff
@@ -58,6 +58,7 @@ pub struct EngineBuilder {
 pub struct ConnectedEngineNative {
     pub config_dir: PathBuf,
     pub env: HashMap<String, String>,
+    #[cfg(all(not(target_os = "ios"), not(target_os = "android")))]
```

Comment:
This target does not support metrics

Reply:
Yes, we leave that for later.

Reply:
This comment is about adding a `metrics` _cargo_ feature to the crate so that instead of specifying

```
#[cfg(all(not(target_os = "ios"), not(target_os = "android")))]
```

you could specify👇 instead

```
#[cfg(feature = "metrics)]
```

Reply:
This is just to avoid having to specify the specific target compilation and abstract it under a cargo feature instead

Reply:
Oh, I get it now. Do you want me to do it in this PR?

Reply:
Yes please, it should be trivial and shouldn't add too much noise. Don't forget to use the feature where needed